pkgs/tools/audio/beets/default.nix

```diff
@@ -217,11 +224,7 @@ in pythonPackages.buildPythonApplication rec {
   '' + optionalString enableConvert ''
     sed -i -e 's,\(util\.command_output(\)\([^)]\+\)),\1[b"${ffmpeg_3.bin}/bin/ffmpeg" if args[0] == b"ffmpeg" else args[0]] + \2[1:]),' beetsplug/convert.py
   '' + optionalString enableReplaygain ''
-    sed -i -re '
-      s!^( *cmd *= *b?['\'''"])(bs1770gain['\'''"])!\1${bs1770gain}/bin/\2!
-    ' beetsplug/replaygain.py
-    sed -i -e 's/if has_program.*bs1770gain.*:/if True:/' \
-      test/test_replaygain.py
+    sed -i -e 's,"ffmpeg","${ffmpeg_3.bin}/bin/ffmpeg",' beetsplug/replaygain.py
```

Comment:
How about removing this part, and adding the patch for replaygain above, with:

```nix
  (substituteAll {
    src = ./ffmpeg-replay-gain.patch;
    ffmpeg = lib.getBin ffmpeg;
  })
```

Ideally perhaps the patch should be applied conditionally if replaygain is enabled.

Reply:
Did the `substituteAll`. I dislike conditional patching, so did not do that.